### DISCUSSION TITLE
CSV Example

### DISCUSSION THREAD

**Discussion by yesoreyeram** [grafana/grafana-infinity-datasource#36] [Show and tell]
## CSV Example

CSV URL : https://thingspeak.com/channels/38629/feed.csv

In the below example, we are using CSV file from thingspeak traffic analysis. As you can see in the screenshot, you can visualise CSV as a table with just URL. 

![image](https://user-images.githubusercontent.com/153843/108428461-8465da00-7236-11eb-8769-b1c145cbe203.png)

Sample data from the above csv

```csv
created_at,entry_id,field1,field2
2021-02-18 21:46:23 UTC,10458189,6.000000,12.000000
2021-02-18 21:46:39 UTC,10458190,0.000000,36.000000
2021-02-18 21:46:55 UTC,10458191,0.000000,49.000000
2021-02-18 21:47:10 UTC,10458192,2.000000,34.000000
2021-02-18 21:47:26 UTC,10458193,6.000000,21.000000
2021-02-18 21:47:41 UTC,10458194,11.000000,13.000000
2021-02-18 21:47:57 UTC,10458195,25.000000,3.000000
2021-02-18 21:48:12 UTC,10458196,29.000000,6.000000

```

## CSV With fields specification

From the same data above, by adding columns and its type, we are able to convert CSV table into a graph. For a graph or any time series visualization, you need a time column and one or more number columns.

![image](https://user-images.githubusercontent.com/153843/108428886-2ab1df80-7237-11eb-8024-8aaab48a6c78.png)

## CSV without time fields

URL : https://gist.githubusercontent.com/yesoreyeram/64a46b02f0bf87cb527d6270dd84ea47/raw/51f2a5e4fe7c3d010a3fe4ae4b6d07961b2ab047/population.csv

![image](https://user-images.githubusercontent.com/153843/108429819-639e8400-7238-11eb-8757-785e29a2394e.png)

In the above example, we don't have any time field. We have a string field and a number field. In this case, by choosing format as timeseries we are simulating the point-in-time as timeseries data. With this option, you can use csv into visualizations like Bar Guage, Stats Panel, Guage panel etc.

Sample Data

```csv
Country,Population
India,3000
China,3500
UK,1200
USA,2000
Germany,700
```

**Comment by ashuvaid** [comment]
Hi,

Thanks for developing the plugin. However request your advise to resolve the error encountered while using the plugin to build a dashboard
I am using docker for desktop and using separate nginx and grafana containers to develop an application
I am using a nginx docker container to host the csv files so as to get a http url to access that file.
I have a separate grafana docker container also running where i have used the Infinity plugin.
In the url section i specify the url http://<private-container-ip>:/port.csv. But the plugin throws error. Error snippet attached for reference

![Capture](https://user-images.githubusercontent.com/67621435/120772720-399a7480-c53e-11eb-839e-2d287b01267e.JPG)


I am able to ping between the nginx and grafana containers. Few logs from my docker from desktop setup

Ping from grafana to nginx container -->
bash-5.0# ping 172.17.0.3
PING 172.17.0.3 (172.17.0.3): 56 data bytes
64 bytes from 172.17.0.3: seq=0 ttl=64 time=0.121 ms
64 bytes from 172.17.0.3: seq=1 ttl=64 time=0.248 ms
64 bytes from 172.17.0.3: seq=2 ttl=64 time=0.120 ms
^C
--- 172.17.0.3 ping statistics ---
3 packets transmitted, 3 packets received, 0% packet loss
round-trip min/avg/max = 0.120/0.163/0.248 ms

Ping from nginx to grafana container -->
root@c047b71a0d80:/data# ping 172.17.0.2
PING 172.17.0.2 (172.17.0.2) 56(84) bytes of data.
64 bytes from 172.17.0.2: icmp_seq=1 ttl=64 time=0.074 ms
64 bytes from 172.17.0.2: icmp_seq=2 ttl=64 time=0.101 ms
64 bytes from 172.17.0.2: icmp_seq=3 ttl=64 time=0.068 ms
^C
--- 172.17.0.2 ping statistics ---
3 packets transmitted, 3 received, 0% packet loss, time 76ms
rtt min/avg/max/mdev = 0.068/0.081/0.101/0.014 ms
root@c047b71a0d80:/data#


$ docker ps
CONTAINER ID   IMAGE             COMMAND                  CREATED          STATUS          PORTS                            NAMES
c047b71a0d80   nginx:latest      "/docker-entrypoint.…"   47 minutes ago   Up 47 minutes   80/tcp, 0.0.0.0:8081->8081/tcp   ngnix_file_server
4acd4f78c9d6   grafana/grafana   "/run.sh"                46 hours ago     Up 19 hours     0.0.0.0:3000->3000/tcp           musing_lewin

**Comment by yosiasz** [comment]
how can one use these https://grafana.com/docs/plugins/yesoreyeram-infinity-datasource/latest/query/macros/#time-macros-__timefrom-and-__timeto

to do time ranges from to using either backend parser or UQL?

```
Date,Total Energy Generation,Total Energy Production,Total Gas Consumption,Total Production,Total Energy Export,PAF,PLF
2024-11-01,704,700,5349,254,4,90,82.8
2024-11-02,706,705,5345,277,1,93,88.35
2024-11-03,706,701,5474,264,5,94,85.54
2024-11-04,708,704,5347,252,4,95,85.5
2024-11-05,714,703,5248,255,11,90,88.2
2024-11-06,718,702,5352,267,16,90,81.9
2024-11-07,720,701,5368,259,19,92,85.56
2024-11-08,715,701,5205,258,14,92,82.8
2024-11-09,709,703,5464,274,6,90,82.8
2024-11-10,716,701,5249,254,15,93,86.49
2024-11-11,707,702,5212,259,5,91,85.54
2024-11-12,707,701,5294,266,6,92,87.4
```
something like `where "Date" between ${__timeFrom} and ${__timeTo}`
Thanks

**Reply by gcbaptista** [reply]
@yesoreyeram could you please take a look at the @yosiasz answer? I've also that doubt.
